tobias/utils/motifs.py

```diff
@@ -122,28 +123,23 @@ def from_file(self, path):
 			proba_flag = False  # read letter-probabilities
 			new_motif = True 	# initialize vars for new motif
 			
-			# init global vars for file
-			bases = ["A", "C", "G", "T"]	#default DNA alphabet
-			strand = "+ -"	#default meme strands
-			bg = np.array([0.25] * 4)
-
+			#Read meme input line by line
 			lines = content.split("\n")
 			for line in lines:
-				# init/ reset motif vars
+
+				# init/ reset current motif
 				if new_motif:
 					new_motif = False
-
-					n = 20 # defined as default number of sites in MEME format
-					info = {}
 					probability_matrix = []
+					self.append(OneMotif(motifid="", counts=[[],[],[],[]]))	#initialize dummy OneMotif for filling in

```

Comment:
It is not possible to create an empty OneMotif object. So this line will throw an error. If you really want this functionality we could say that counts is `None` as default but I would advise against it. Because if we accept empty or partly empty Objects we need to think about what for example `get_pfm` does when there is no counts matrix.

Reply:
Also, I can see why you want to create an empty object and fill it. But in this case, doing it like this will result in an empty OneMotif object as the last element of each MotifList.